docs/modules/hdfs/pages/usage-guide/security.adoc

```diff
@@ -54,25 +54,61 @@ In case you want to access your HDFS it is recommended to start up a client Pod
-        hadoop.user.group.static.mapping.overrides: "dr.who=;nn=;nm=;jn=;testuser=supergroup;"
-  dataNodes:
-    configOverrides: *configOverrides
-  journalNodes:
-    configOverrides: *configOverrides
+  clusterConfig:
+    authorization:
+      opa:
+        configMapName: opa
+        package: hdfs
 ----
 
+=== How it works
+WARNING: Take all your knowledge about HDFS authorization and throw it in the bin.
+The approach we are taking for our authorization paradigm departs significantly from traditional Hadoop patterns and POSIX-style permissions.
+
+To put it short, the current regorules ignore the file ownership, permissions, ACLs and all other attributes files can have.
+All of this is state in HDFS and clashes with the infrastructure as code approach (IaC).
+
+Instead HDFS will send a request who (e.g. `alice/test-hdfs-permissions.default.svc.cluster.local@CLUSTER.LOCAL`) is trying to do what (e.g. `open`, `create`, `delete` or `append`) on what file (e.g. `/foo/bar`).
+OPA than makes a decision if this action is allowed or not.
+
+Instead of `chown` ing a directory do a different user to give him write permissions, what you really should do is go to your IaC Git repository and add a rego-rule entry that the user is allowed to read and write to that directory.
+
+=== Group memberships
+While implementing the group mapper, we encountered numerous problems with it, with the biggest problem being that the `GroupMappingServiceProvider` interface only passes the `shortUsername` when https://github.com/apache/hadoop/blob/a897e745f598ef05fc0c253b2a776100e48688d2/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/GroupMappingServiceProvider.java#L45[asking for group memberships].
+This does not allow us to differentiate e.g. `hbase/hbase-prod.prod-namespace.svc.cluster.local@CLUSTER.LOCAL` and `hbase/hbase-dev.dev-namespace.svc.cluster.local@CLUSTER.LOCAL`, as the GroupMapper will get asked for `hbase` group memberships in both cases.
+
+Users could work around this by giving users unique shortNames by using https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/SecureMode.html#Mapping_from_Kerberos_principals_to_OS_user_accounts[`hadoop.security.auth_to_local`].
+This is however a potentially complex and error-prone process.
+We also tried mapping the principals from Kerberos 1:1 to the HDFS `shortUserName`, so the GroupMapper has access to the full `userName`.
+However, this did not work, as HDFS only allows "simple usernames", https://github.com/apache/hadoop/blob/8378ab9f92c72dc6164b62f7be71826fd750dba4/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/util/KerberosName.java#L348[which are not allowed to contain a `/` or `@`].
+
+Because of the mentioned problems we stopped using a GroupMapper and only rely on the authorizer, which in turn get's the whole `UserGroupInformation`, including the `shortUserName` and the precious full `userName`.
```

Comment:
```suggestion
Because of these issues we do not use a custom GroupMapper and only rely on the authorizer, which in turn receives a complete `UserGroupInformation` object, including the `shortUserName` and the precious full `userName`.
```